osprey/processes/wps_parameters.py

```diff
@@ -0,0 +1,130 @@
+# Processor imports
+from pywps import (
+    Process,
+    LiteralInput,
+)
+from pywps.app.Common import Metadata
+
+# Tool imports
+from rvic.parameters import parameters
+from rvic.core.config import read_config
+from wps_tools.utils import (
+    collect_output_files,
+    log_handler,
+)
+from wps_tools.io import (
+    log_level,
+    nc_output,
+)
+
+# Library imports
+import logging
+import os
+
+pywps_logger = logging.getLogger("PYWPS")
+
+
+class Parameters(Process):
+    def __init__(self):
+        self.status_percentage_steps = {
+            "start": 0,
+            "dry_run": 5,
+            "process": 10,
+            "collect_files": 90,
+            "build_output": 95,
+            "complete": 100,
+        }
+        inputs = [
+            LiteralInput(
+                "config",
+                "Configuration",
+                abstract="Path to input configuration file or input dictionary",
+                data_type="string",
+            ),
+            LiteralInput(
+                "np",
+                "numofproc",
+                default=1,
+                abstract="Number of processors used to run job",
+                data_type="integer",
+            ),
+            LiteralInput(
+                "version",
+                "Version",
+                default=True,
+                abstract="Return RVIC version string",
+                data_type="boolean",
+            ),
+        ]
+        outputs = [
+            nc_output,
+        ]
+
+        super(Parameters, self).__init__(
+            self._handler,
+            identifier="parameters",
+            title="Parameters",
+            abstract="Develop impulse response functions using inputs from a "
+            "configuration file or dictionary",
+            metadata=[
+                Metadata("NetCDF processing"),
+                Metadata("Climate Data Operations"),
+            ],
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def collect_args(self, request):
+        config = request.inputs["config"][0].data
+        np = request.inputs["np"][0].data
+        return (config_file, np)
+
+    def get_outfile(self, config):
+        if isinstance(config, dict):
+            config_dict = config
+        else:
+            config_dict = read_config(config)
+        outdir = os.path.join(config_dict["OPTIONS"]["CASE_DIR"], "params")
+        (param_file,) = collect_output_files("rvic", outdir)
+        return os.path.join(outdir, param_file)
+
+    def _handler(self, request, response):
+        if request.inputs["version"][0].data:
+            from rvic import version
+
+            pywps_logger.info(version.short_version)
+
+        (config, np) = self.collect_args(request)
+        log_handler(
+            self, response, "Starting Process", process_step="start", level=loglevel
+        )
+        if config[0] != "{":
```

Comment:
Can you explain what you are looking for here?

Reply:
I'm checking to see if the first character is a `{` since that shows that the input is a dictionary rather than a file path. I'll clarify that.

Reply:
I think there is a better way to go about that. We can use `type(input) == dict` to check if it is a dict. If it comes in as a `string` in the format of a `dict` I think there is a way to translate that into a `dict`.

Reply:
On second thought, maybe I could `try` to run `eval(config)`, and if it raises an exception, check to make sure it's a file path in the `except` block.